### ISSUE TITLE
Topics TODO

### ISSUE THREAD

**Issue by ChJus** [JC-ProgJava/Building-Neural-Networks-From-Scratch#28]
### Evaluation
- [ ] [Confusion matrices](https://towardsdatascience.com/understanding-confusion-matrix-a9ad42dcfd62)
  - [Source 2](https://en.wikipedia.org/wiki/Confusion_matrix)
- [ ] [Precision and recall](https://en.wikipedia.org/wiki/Precision_and_recall)

### Regression / Loss/Cost Function
- [ ] Polynomial Regression (put into Chapter 5)
- [ ] Ridge & LASSO Regression
- [ ] Elastic Net Regression
- [x] Logistic (Probabilistic) Regression

### Regularization Techniques
- [ ] [Early Stopping](https://en.wikipedia.org/wiki/Early_stopping)
- [x] [Dropout](https://en.wikipedia.org/wiki/Dilution_(neural_networks))
- [ ] [Weight Decay](https://towardsdatascience.com/this-thing-called-weight-decay-a7cd4bcfccab) (L2 Regularization)

### Other Architectures (Appendix and New Chapters)
- Neural Networks
  - [ ] [Recurrent Neural Networks (RNN)](https://en.wikipedia.org/wiki/Recurrent_neural_network)
  - [ ] [Auto-encoders](https://en.wikipedia.org/wiki/Autoencoder)
  - [ ] [Generative Adversarial Networks (GAN)](https://en.wikipedia.org/wiki/Generative_adversarial_network)
- [ ] [Support Vector Machines (SVM)](https://en.wikipedia.org/wiki/Support-vector_machine)
- [ ] [Decision Trees](https://en.wikipedia.org/wiki/Decision_tree_learning)
- [Reinforcement Learning](https://en.wikipedia.org/wiki/Reinforcement_learning)
  - [ ] [Q-Learning](https://en.wikipedia.org/wiki/Q-learning)
- [Unsupervised Learning](https://en.wikipedia.org/wiki/Unsupervised_learning)
  - [ ] [K-Means](https://en.wikipedia.org/wiki/K-means_clustering)
  - [ ] [K-Nearest Neighbor](https://en.wikipedia.org/wiki/K-nearest_neighbors_algorithm)
- [ ] [Ensemble Learning](https://en.wikipedia.org/wiki/Ensemble_learning)
  - [ ] [Random Forests](https://en.wikipedia.org/wiki/Random_forest)

### Preprocessing
- [ ] [Dimensionality Reduction](https://en.wikipedia.org/wiki/Dimensionality_reduction)

**Comment by ChJus** [comment]
- [ ] Update 'A Readable Bibliography'
- [ ] Further Reading
